File: ios/Packages/Components/Sources/Components/SheetActionItem.swift
import DesignSystem
import SwiftUI

public struct SheetActionItem: View {
    @Environment(\.appTheme) private var theme

    private let icon: String
    private let label: String
    private let supportingText: String?
    private let isEnabled: Bool
    private let showProgress: Bool
    private let action: () -> Void

    public init(
        icon: String,
        label: String,
        supportingText: String? = nil,
        isEnabled: Bool = true,
        showProgress: Bool = false,
        action: @escaping () -> Void
    ) {
        self.icon = icon
        self.label = label
        self.supportingText = supportingText
        self.isEnabled = isEnabled
        self.showProgress = showProgress
        self.action = action
    }

    public var body: some View {
        Button(action: action) {
            HStack(spacing: theme.spacing.medium) {
                ZStack {
                    if showProgress {
                        ProgressView()
                            .controlSize(.small)
                            .tint(theme.colors.onSurfaceVariant)
                    } else {
                        Image(systemName: icon)
                            .textStyle(theme.typography.titleMedium)
                            .foregroundStyle(.appOnSurface)
                            .opacity(isEnabled ? 1 : 0.38)
                    }
                }
                .frame(width: 24)

                VStack(alignment: .leading, spacing: 0) {
                    Text(label)
                        .textStyle(theme.typography.bodyLarge)
                        .foregroundStyle(.appOnSurface)
                        .opacity(isEnabled ? 1 : 0.38)

                    if let supportingText {
                        Text(supportingText)
                            .textStyle(theme.typography.bodySmall)
                            .foregroundStyle(theme.colors.onSurfaceVariant)
                            .opacity(isEnabled ? 1 : 0.38)
                    }
                }

                Spacer()
            }
            .padding(.horizontal, theme.spacing.medium)
            .padding(.vertical, theme.spacing.small)
        }
        .buttonStyle(PlainButtonStyle())
        .disabled(!isEnabled)
    }
}

#Preview("Sheet action item") {
    VStack(spacing: 0) {
        SheetActionItem(icon: "checkmark.circle", label: "Just now", action: {})
        SheetActionItem(icon: "clock", label: "Release date", isEnabled: false, action: {})
        SheetActionItem(
            icon: "calendar",
            label: "Change watched date",
            supportingText: "12 Jan 2026 20:30",
            action: {}
        )
    }
}

File: android-designsystem/src/main/kotlin/com/thomaskioko/tvmaniac/compose/components/SheetActionItem.kt
package com.thomaskioko.tvmaniac.compose.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Check
import androidx.compose.material.icons.outlined.Schedule
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.tooling.preview.PreviewWrapper
import androidx.compose.ui.unit.dp
import com.thomaskioko.tvmaniac.compose.theme.TvManiacSpacing

@Composable
public fun SheetActionItem(
    icon: ImageVector,
    label: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    showProgress: Boolean = false,
    supportingText: String? = null,
) {
    val contentAlpha = if (enabled) 1f else 0.38f
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clickable(enabled = enabled, onClick = onClick)
            .padding(horizontal = TvManiacSpacing.medium, vertical = TvManiacSpacing.small),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(TvManiacSpacing.medium),
    ) {
        Box(
            modifier = Modifier.size(24.dp),
            contentAlignment = Alignment.Center,
        ) {
            if (showProgress) {
                CircularProgressIndicator(
                    modifier = Modifier.size(20.dp),
                    strokeWidth = 2.dp,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            } else {
                Icon(
                    imageVector = icon,
                    contentDescription = null,
                    modifier = Modifier.size(24.dp),
                    tint = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = contentAlpha),
                )
            }
        }
        Column {
            Text(
                text = label,
                style = MaterialTheme.typography.bodyLarge,
                color = MaterialTheme.colorScheme.onSurface.copy(alpha = contentAlpha),
            )
            if (supportingText != null) {
                Text(
                    text = supportingText,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = contentAlpha),
                )
            }
        }
    }
}

@ThemePreviews
@PreviewWrapper(TvManiacPreviewWrapperProvider::class)
@Composable
private fun SheetActionItemPreview() {
    SheetActionItem(
        icon = Icons.Outlined.Check,
        label = "Just now",
        onClick = {},
    )
}

@ThemePreviews
@PreviewWrapper(TvManiacPreviewWrapperProvider::class)
@Composable
private fun SheetActionItemDisabledPreview() {
    SheetActionItem(
        icon = Icons.Outlined.Schedule,
        label = "Release date",
        onClick = {},
        enabled = false,
    )
}

@ThemePreviews
@PreviewWrapper(TvManiacPreviewWrapperProvider::class)
@Composable
private fun SheetActionItemSupportingTextPreview() {
    SheetActionItem(
        icon = Icons.Outlined.Schedule,
        label = "Change watched date",
        onClick = {},
        supportingText = "12 Jan 2026 20:30",
    )
}
